pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -456,101 +453,69 @@ func (nim *nodeInfoManager) installDriverToCSINodeInfo(
 			newDriverSpecs = append(newDriverSpecs, driverInfoSpec)
 		}
 	}
-	newDriverStatuses := []csiv1alpha1.CSIDriverInfoStatus{}
-	for _, driverInfoStatus := range nodeInfo.Status.Drivers {
-		if driverInfoStatus.Name == driverName {
-			if driverInfoStatus.Available &&
-				/* TODO(https://github.com/kubernetes/enhancements/issues/625): Add actual migration status */
-				driverInfoStatus.VolumePluginMechanism == csiv1alpha1.VolumePluginMechanismInTree {
-				statusModified = false
-			}
-		} else {
-			// Omit driverInfoSpec matching given driverName
-			newDriverStatuses = append(newDriverStatuses, driverInfoStatus)
-		}
-	}
 
 	if !specModified && !statusModified {
 		return nil
 	}
 
 	// Append new driver
-	driverSpec := csiv1alpha1.CSIDriverInfoSpec{
+	driverSpec := storage.CSINodeDriver{
 		Name:         driverName,
 		NodeID:       driverNodeID,
 		TopologyKeys: topologyKeys.List(),
 	}
-	driverStatus := csiv1alpha1.CSIDriverInfoStatus{
-		Name:      driverName,
-		Available: true,
-		// TODO(https://github.com/kubernetes/enhancements/issues/625): Add actual migration status
-		VolumePluginMechanism: csiv1alpha1.VolumePluginMechanismInTree,
-	}
 
 	newDriverSpecs = append(newDriverSpecs, driverSpec)
-	newDriverStatuses = append(newDriverStatuses, driverStatus)
 	nodeInfo.Spec.Drivers = newDriverSpecs
-	nodeInfo.Status.Drivers = newDriverStatuses
 
-	err := validateCSINodeInfo(nodeInfo)
+	err := validateCSINode(nodeInfo)
 	if err != nil {
 		return err
 	}
-	_, err = csiKubeClient.CsiV1alpha1().CSINodeInfos().Update(nodeInfo)
+	_, err = csiKubeClient.StorageV1beta1().CSINodes().Update(nodeInfo)
 	return err
 }
 
-func (nim *nodeInfoManager) uninstallDriverFromCSINodeInfo(
+func (nim *nodeInfoManager) uninstallDriverFromCSINode(
```

Comment:
The original logic also [deleted the CSINode](https://github.com/kubernetes/kubernetes/commit/06f3b26012e09318935c248e62864e9349da6cd8#diff-0e13ddc214dda9ec6a0a089c62d372aeL482) object when no drivers remained.  We should add that back. @davidz627 @verult were there any potential races with register/unregister?